bite-size-notes/Week-3-d-Optimal Control-Jiangwen Cheng.md

```diff
@@ -1 +1,130 @@
+
+$$
+\begin{align}
+Q = \{ x \in D \subset \mathbb {R^n} : b(x) \geq 0 \}, 
+\\
+\partial Q = \{x \in D \subset \mathbb {R^n} : b(x) = 0 \}, 
+\\
+\text{Int}(Q) = \{x \in D \subset \mathbb {R^n} : b(x) > 0 \}. 
+\end{align}
+$$  
+
+We refer to ${Q}$ as the safe set.   
+
+While inspired by and expanding upon Control Lyapunov Functions (CLFs), Control Barrier Functions (CBFs) differ by aiming to enforce set invariance without the strict requirement of a positive definite function. The continuously differentiable function defined for CBFs ensures the invariance of our designated safe set, but not its sublevel sets.
+This motivates the formulation of control barrier functions. Before defining these, we note that an extended class ${K_\infty}$ function is a function $\alpha : \mathbb{R} \longrightarrow \mathbb{R}$ that is strictly increasing and with $\alpha(0) =0$. Extended functions are defined on entire real line: $\mathbb R = (-\infty, \infty)$. Then, we can define the Control Barrier Functions now.   
+
+### Definition (Control Barrier Function)
+
+Let $Q \subset D \subset \mathbb {R^n}$ be the superlevel set of a continuously differentiable function $b : D \longrightarrow \mathbb {R}$, then $b$ is a Control Barrier Function (CBF) if there exists an extended class $K_{\infty}$ function $\alpha$ such that for the control affine system: [3]
+
+$$
+\text{sup}_{u\in U} [L_f b(x) + L_g b(x) u] \geq - \alpha (b(x)) $$ 
+
+for all $x \in D$  
+Hence, we can quantify the set of all control inputs at a point $x \in D$ that can guarantee the system's safety via CBFs. 
+
+$$
+K_{cbf}(x) = \{u \in U : L_f b(x) + L_g b(x) u + \alpha (b(x)) \geq 0 \}
+$$  
+
+Common choice of $\alpha$ is to be $:\alpha(r) = \alpha_0(r)$
+
+To better understand the Control Barrier Function, we examine the following cases for the system as it moving to the undesirable set ${C}$. ${C}$ is unsafe region where $C = \{x \in D \mid b(x) \leq 0 \}$ and assume $\alpha_0 =1$. 
+
+1. The system is relatively far from the undesired set so it can to move in any direction basically. Notice not all directions are allowed such as going directly toward C. 
+
+$$
+L_f b(x) + L_g b(x) u   \geq - 3
+$$
+
+
+![alt text](figs/1.png)
+
+2. As the system approaches the undesired set ${C}$, the range of available directions to take is reduced. 
+
+$$ 
+L_f b(x) + L_g b(x) u   > - 1 
+$$
+
+![alt text](figs/2.png)
+
+3. When the system is on the boundary of the undesired set, the only possible set of actions are to move along the boundary of ${C}$ or to move away in any direction.
+
+$$ 
+L_f b(x) + L_g b(x) u   \geq 0 
+$$
+
+![alt text](figs/3.png)
+
+4. When the system moves away from the undesired set, the possible range of actions in increased. 
+
+$$
+L_f b(x) + L_g b(x) u   \geq -1 
+$$
+
+![alt text](figs/4.png)
+
+## Optimization Based Control 
```

Comment:
The write up gives a good enough understanding about the topic. Kudos. 
You might want to add the general setup of Optimal Control Problem for both discrete and continuous time with explanation of Terminal and running cost. 
It would also give an idea about Dynamic Programming in future notes. 

There are also a few typos here which you might want to correct.

Reply:
Since my topic is mainly about the CBF, I'm not going to take optimal control problem in this note. Thanks!